### PR TITLE
#37 edit article

### PR DESCRIPTION
**Pull request by martuto31**
Implemented functionality for editing an article.

### CODE REVIEW

File: SportsHub/SportsHub.Api/Controllers/ArticleController.cs [hunk 1]
```diff
@@ -64,5 +64,26 @@ public async Task<IActionResult> CreateArticleAsync([FromBody] CreateArticleDTO
...
+        public async Task<IActionResult> EditArticleAsync([FromBody] CreateArticleDTO adminInput)
+        {
+            ValidationResult validationResult = await _articleValidator.ValidateAsync(adminInput);
+            if (!validationResult.IsValid)
+            {
+                var response = _generateModelStateDictionary.modelStateDictionary(validationResult);
+                return ValidationProblem(response);
+            }
+
+            bool editedSuccessfylly = await _articleService.EditArticle(adminInput);
```
Comment (MBocharskyi) [inline SportsHub/SportsHub.Api/Controllers/ArticleController.cs:79]: Typo. editedSuccessf**u**lly

File: SportsHub/SportsHub.Api/Controllers/ArticleController.cs [hunk 2]
```diff
@@ -64,5 +64,26 @@ public async Task<IActionResult> CreateArticleAsync([FromBody] CreateArticleDTO
...
+
+            if (editedSuccessfylly)
+            {
+                return Ok("Article updated successfully.");
```
Comment (MBocharskyi) [inline SportsHub/SportsHub.Api/Controllers/ArticleController.cs:83]: Please do not use magic strings.

**Review by MBocharskyi — CHANGES_REQUESTED** [review]

File: SportsHub/SportsHub.Api/Controllers/ArticleController.cs (hunk 1 above)
Comment (martuto31) [reply SportsHub/SportsHub.Api/Controllers/ArticleController.cs]: The typo is fixed.

File: SportsHub/SportsHub.Api/Controllers/ArticleController.cs (hunk 2 above)
Comment (martuto31) [reply SportsHub/SportsHub.Api/Controllers/ArticleController.cs]: Object result messages are now in ValidationMessages.

File: SportsHub/SportsHub.Api/Controllers/ArticleController.cs [hunk 3]
```diff
@@ -64,5 +65,26 @@ public async Task<IActionResult> CreateArticleAsync([FromBody] CreateArticleDTO
...
+
+        [Authorize(Roles = "Admin")]
+        [HttpPut("EditArticle")]
+        public async Task<IActionResult> EditArticleAsync([FromBody] CreateArticleDTO adminInput)
+        {
+            ValidationResult validationResult = await _articleValidator.ValidateAsync(adminInput);
+            if (!validationResult.IsValid)
+            {
+                var response = _generateModelStateDictionary.modelStateDictionary(validationResult);
+                return ValidationProblem(response);
```
Comment (NataliiaBidiuk) [inline SportsHub/SportsHub.Api/Controllers/ArticleController.cs:77]: add line everywhere before Return, please)

File: SportsHub/SportsHub.Application/Services/ArticleService.cs [hunk 4]
```diff
@@ -43,5 +43,22 @@ public async Task<bool> CreateArticleAsync(CreateArticleDTO adminInput)
...
+        public async Task<bool> EditArticle(CreateArticleDTO adminInput)
+        {
+            var article = GetByTitleAsync(adminInput.Title).GetAwaiter().GetResult();
+
+            if(article == null)
+            {
+                return false;
+            }
+
+            article.CategoryId = adminInput.CategoryId;
```
Comment (NataliiaBidiuk) [inline SportsHub/SportsHub.Application/Services/ArticleService.cs:56]: add this into mapper, please

File: SportsHub/SportsHub.Core/Models/Constants/ValidationMessages.cs [hunk 5]
```diff
@@ -0,0 +1,8 @@
+﻿namespace SportsHub.Domain.Models.Constants
+{
+    public static class ValidationMessages
+    {
+        public const string ArticleUpdatedSuccessfully = "Article was updated successfully.";
+        public const string ArticleNotFound = "Article not found.";
```
Comment (NataliiaBidiuk) [inline SportsHub/SportsHub.Core/Models/Constants/ValidationMessages.cs:6]: add Unit tests, please

File: SportsHub/SportsHub.Application/Services/ArticleService.cs [hunk 6]
```diff
@@ -43,5 +43,22 @@ public async Task<bool> CreateArticleAsync(CreateArticleDTO adminInput)
 
             return true;
         }
+
+        public async Task<bool> EditArticle(CreateArticleDTO adminInput)
+        {
+            var article = GetByTitleAsync(adminInput.Title).GetAwaiter().GetResult();
```
Comment (NataliiaBidiuk) [inline SportsHub/SportsHub.Application/Services/ArticleService.cs:49]: why by title?

**Review by MBocharskyi — APPROVED** [review]

File: SportsHub/SportsHub.Api/Controllers/ArticleController.cs (hunk 3 above)
Comment (martuto31) [reply SportsHub/SportsHub.Api/Controllers/ArticleController.cs:95]: Added.

File: SportsHub/SportsHub.Application/Services/ArticleService.cs (hunk 6 above)
Comment (martuto31) [reply SportsHub/SportsHub.Application/Services/ArticleService.cs]: Changed to search by Id.

File: SportsHub/SportsHub.Application/Services/ArticleService.cs (hunk 4 above)
Comment (martuto31) [reply SportsHub/SportsHub.Application/Services/ArticleService.cs]: Added auto mapper and check if the CategoryId exists.

**Review by NataliiaBidiuk — APPROVED** [review]

**Comment by NataliiaBidiuk** [comment]
Please, resolve conflicts, and adjust result by using ExceptionHandler, get rid of true and false)

**Pull request merged**